src/Microsoft.DotNet.Darc/src/DarcLib/VirtualMonoRepo/ThirdPartyNoticesGenerator.cs

```diff
@@ -46,11 +46,10 @@ public async Task UpdateThirtPartyNotices()
     {
         _logger.LogInformation("Updating {tpnName}...", VmrInfo.ThirdPartyNoticesFileName);
 
-        var templatePath = _vmrInfo.VmrPath / VmrInfo.ThirdPartyNoticesTemplatePath;
         string header = string.Empty;
-        if (_fileSystem.FileExists(templatePath))
+        if (_vmrInfo.ThirdPartyNoticesTemplatePath != null && _fileSystem.FileExists(_vmrInfo.ThirdPartyNoticesTemplatePath))
         {
-            header = await _fileSystem.ReadAllTextAsync(templatePath);
+            header = await _fileSystem.ReadAllTextAsync(_vmrInfo.ThirdPartyNoticesTemplatePath);
```

Comment:
Shouldn't we return at the top of the method if `null`? Seems like we would still run the generation but without the header which is not what we want. We'd like to skip the generation altogether.

Reply:
ah ok

Reply:
then it needs to be a required argument for `generate-tpn` command